src/qt/transactiontablemodel.cpp

```diff
@@ -64,21 +66,38 @@ class TransactionTablePriv
 
     /* Query entire wallet anew from core.
      */
-    void refreshWallet()
+    void loadWallet()
     {
-        LogPrint(BCLog::LogFlags::VERBOSE, "refreshWallet");
         cachedWallet.clear();
         {
             LOCK2(cs_main, wallet->cs_wallet);
+
+            bool fLimitTxnDisplay = walletModel->getOptionsModel()->getLimitTxnDisplay();
+            int64_t limitTxnDateTime = walletModel->getOptionsModel()->getLimitTxnDateTime();
+
             for(std::map<uint256, CWalletTx>::iterator it = wallet->mapWallet.begin(); it != wallet->mapWallet.end(); ++it)
             {
-                if(TransactionRecord::showTransaction(it->second))
+                if (TransactionRecord::showTransaction(it->second, fLimitTxnDisplay, limitTxnDateTime))
+                {
                     cachedWallet.append(TransactionRecord::decomposeTransaction(wallet, it->second));
+                }
             }
         }
     }
 
-    /* Update our model of the wallet incrementally, to synchronize our model of the wallet
+
+    /* Update QList using new query from core.
```

Comment:
Is there still a core API for this?

Reply:
No, other than walking the wallet map. Maybe the comment is confusing.